github/TimelineEvent.py

```diff
@@ -0,0 +1,159 @@
+# http://pygithub.readthedocs.io/                                              #
+#                                                                              #
+# PyGithub is free software: you can redistribute it and/or modify it under    #
+# the terms of the GNU Lesser General Public License as published by the Free  #
+# Software Foundation, either version 3 of the License, or (at your option)    #
+# any later version.                                                           #
+#                                                                              #
+# PyGithub is distributed in the hope that it will be useful, but WITHOUT ANY  #
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS    #
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more #
+# details.                                                                     #
+#                                                                              #
+# You should have received a copy of the GNU Lesser General Public License     #
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+
+from __future__ import absolute_import
+import github.GithubObject
+
+import github.NamedUser
+import github.TimelineEventSource
+
+
+class TimelineEvent(github.GithubObject.NonCompletableGithubObject):
+    """
+    This class represents IssueTimelineEvents. The reference can be found here https://developer.github.com/v3/issues/timeline/
+    """
+
+    def __repr__(self):
+        return self.get__repr__({"id": self._id.value})
+
+    @property
+    def actor(self):
+        """
+        :type: :class:`github.NamedUser.NamedUser`
+        """
+        return self._actor.value
+
+    @property
+    def commit_id(self):
+        """
+        :type: string
+        """
+        return self._commit_id.value
+
+    @property
+    def created_at(self):
+        """
+        :type: datetime.datetime
+        """
+        return self._created_at.value
+
+    @property
+    def event(self):
+        """
+        :type: string
+        """
+        return self._event.value
+
+    @property
+    def id(self):
+        """
+        :type: integer
+        """
+        return self._id.value
+
+    @property
+    def node_id(self):
+        """
+        :type: string
+        """
+        return self._node_id.value
+
+    @property
+    def commit_url(self):
+        """
+        :type: string
+        """
+        return self._commit_url.value
+
+    @property
+    def source(self):
+        """
+        :type: :class:`github.TimelineEventSource.TimelineEventSource`
+        """
+        # only available on `cross-referenced` events.
+        if self.event == 'cross-referenced' and hasattr(self, '_source') and self._source is not None:
```

Comment:
Actually, this and the below checks are also wrong, because they'll be NotSet, not None. The _hasattr() feels wrong too, surely they're set in initAttributes ...

Reply:
Hmm. At some point I was having issues with that, but it was likely early on in the changes I was making. I think it was that I wasn't setting the properties in `initAttributes`. Thanks.